pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,36 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional OpenShift components are specified in the
+// install-config to be included in the installation.
+func validateCapabilities(c *configv1.ClusterVersionCapabilitiesSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allAvailableCapabilities := map[string]bool{}
+	var allCaps, allSets []string
+	// Create a map and slice of *all* available sets and capabilities across all capability sets
+	for baselineSet, capabilities := range configv1.ClusterVersionCapabilitySets {
+		allSets = append(allSets, string(baselineSet))
+		for _, capability := range capabilities {
+			if _, found := allAvailableCapabilities[string(capability)]; !found {
+				allAvailableCapabilities[string(capability)] = true
+				allCaps = append(allCaps, string(capability))
+			}
+		}
+	}
+
+	if _, found := configv1.ClusterVersionCapabilitySets[c.BaselineCapabilitySet]; !found {
```

Comment:
This will raise an error if `BaselineCapabilitySet` is omitted. We want the field to be optional, right? We should add a test case for this.

Reply:
True. I've added a check to catch empty values.